employee-rostering-frontend/src/ui/Alerts.tsx

```diff
@@ -45,10 +45,8 @@ const mapDispatchToProps: DispatchProps = {
 export type Props = StateProps & DispatchProps;
 
 export function mapToComponent(component: AlertComponent, componentProps: BasicObject): React.ReactNode {
-  switch (component) {
-    case AlertComponent.SERVER_SIDE_EXCEPTION_DIALOG: {
-      return <ServerSideExceptionDialog {...componentProps as unknown as ServerSideExceptionInfo} />
-    }
+  if (component === AlertComponent.SERVER_SIDE_EXCEPTION_DIALOG) {
+    return <ServerSideExceptionDialog {...componentProps as unknown as ServerSideExceptionInfo} />;
   }
```

Comment:
Is "undefined" a `React.ReactNode`? If not, I don't think it will pass a typecheck, unless typescript sees "SERVER_SIDE_EXCEPTION_DIALOG" is the only `AlertComponent`, in which TypeScript should complain about an if that is always true.

Reply:
Good question. Yes it is.
```ts
    type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
```

Reply:
Currently, there's only one `AlertComponent`:
```ts

export enum AlertComponent {
  SERVER_SIDE_EXCEPTION_DIALOG = 'SERVER_SIDE_EXCEPTION_DIALOG'
}
```
but it's still technically possible for the condition to be `false` in runtime.